rusqlite_migration/src/builder.rs

```diff
@@ -42,14 +42,11 @@ impl<'u> MigrationsBuilder<'u> {
     ///
     /// Panics if no migration with the `id` provided exists.
     #[must_use]
-    pub fn edit(mut self, id: usize, f: impl Fn(&mut M)) -> Self {
+    pub fn edit(mut self, id: usize, f: impl Fn(M) -> M) -> Self {
         if id < 1 {
             panic!("id cannot be equal to 0");
         }
-        f(self
-            .migrations
-            .get_mut(id - 1)
-            .expect("No migration with the given index"));
+        self.migrations[id - 1] = f(self.migrations[id - 1].clone());
```

Comment:
Oh you are right, migrations are actually relatively cheap to clone, it’s mostly pointers or booleans.